core/src/main/scripts/migrate_db.py

```diff
@@ -0,0 +1,204 @@
+        print >> ERROR_FILE, 'Missing one or more required properties, please check property file'
+        return none
+    
+    # return an instance of PortalProperties
+    return PortalProperties(properties[DATABASE_HOST],
+                            properties[DATABASE_NAME],
+                            properties[DATABASE_USER],
+                            properties[DATABASE_PW])
+
+def get_db_version(cursor):
+    """ gets the version number of the database """
+
+    # First, see if the version table exists
+    version_table_exists = False
+    try:
+        cursor.execute('select table_name from information_schema.tables')
+        for row in cursor.fetchall():
+            if VERSION_TABLE == row[0].lower().strip():
+                version_table_exists = True
+    except MySQLdb.Error, msg:
+        print >> ERROR_FILE, msg
+        return None
+    
+    if not version_table_exists:
+        return (0,0,0)
+
+    # Now query the table for the version number
+    try:
+        cursor.execute('select ' + VERSION_FIELD + ' from ' + VERSION_TABLE)
+        for row in cursor.fetchall():
+            version = tuple(map(int, row[0].strip().split('.')))
+    except MySQLdb.Error, msg:
+        print >> ERROR_FILE, msg
+        return None
+
+    return version
+
+def is_version_larger(version1, version2):
+    """ Checks if version 1 is larger than version 2 """
+    
+    if version1[0] > version2[0]:
+        return True
+    if version2[0] > version1[0]:
+        return False
+    if version1[1] > version2[1]:
+        return True
+    if version2[1] > version1[1]:
+        return False
+    if version1[2] > version2[2]:
+        return True
+    return False
+
+def run_migration(db_version, sql_filename, cursor):
+    """
+
+        Goes through the sql and runs lines based on the version numbers. SQL version should be stated as follows:
+
+        ##version: 1.0.0
+        INSERT INTO ...
+
+        ##version: 1.1.0
+        CREATE TABLE ...
+    
+    """
+    
+    sql_file = open(sql_filename, 'r')
+    sql_version = (0,0,0)
+    run_line = False
+
+    for line in sql_file:
+        if line.startswith('##'):
+            sql_version = tuple(map(int, line.split(':')[1].strip().split('.')))
+            run_line = is_version_larger(sql_version, db_version)
+            continue
+
+        # skip blank lines
+        if len(line.strip()) < 1 or line.startswith('#'):
+            continue
+        # only execute sql line if the last version seen in the file is greater than the db_version
+        if run_line:
+            try:
+                cursor.execute(line.strip())
+            except MySQLdb.Error, msg:
+                print >> ERROR_FILE, msg
+                sys.exit(1)
+            
+
+def usage():
+    print >> OUTPUT_FILE, 'migrate_db.py --properties-file [portal properties file] --sql [sql migration file]'
+
+def main():
+    """ main function to run mysql migration """
+    
+    try:
```

Comment:
maybe before starting, warn the user that he needs to backup his DB in case something goes wrong...(it can happen that there is still a bug in the migration script)?